tsdb/config.go

```diff
@@ -151,6 +161,8 @@ func NewConfig() Config {
 
 		MaxConcurrentCompactions: DefaultMaxConcurrentCompactions,
 
+		WALMaxWriteDelay: 10 * time.Minute,
```

Comment:
This is setting the default if no specific flag is passed right? If so I'm wondering about 10 minutes....seems kind of long? I don't have much context (pun not intended) on what long or not long is for this, so I'm curious if there is any reasoning to go along with this as the default. Also wondering if it should default to not having a timeout at all, to be consistent with previous behavior.

Reply:
10 minutes is definitely arbitrary, and longer than we'd ever expect a write to take. @lesam suggested it as a value that we can be very confident signals something has locked up / is overloaded. We could probably use a smaller value like 5 minutes (or even 1 or 2 minutes?) but I don't have a clear sense of how to judge what the minimum "safe" value is.

I think we want to default to having some timeout here because the previous behavior (in 2.0.x) was to allow an arbitrary number of concurrent writes, and adding the limiter already introduced a behavior change by making some writes lock up (or appear to). This follow-up doesn't completely restore the old behavior, but I think it makes the failure case nicer for the user by default (getting a timeout error instead of wondering why the request isn't completing).

Reply:
Makes sense to me!